gpkit/constraints/geometric_program.py

```diff
@@ -271,28 +252,20 @@ def _compile_result(self, solver_out):
 
         ## Get sensitivities
         result["sensitivities"] = {"nu": nu, "la": la}
-        # initialize the var_senss dict with the cost's constants...
-        var_senss = {var: sum([self.cost.exps[i][var]*nu[i] for i in locs])
-                     for (var, locs) in self.cost.varlocs.items()
-                     if (var in self.cost.varlocs
-                         and var not in self.posynomials[0].varlocs)}
-        var_senss = HashVector(var_senss)
-        for c_i, posy_idxs in enumerate(self.constr_idxs):
-            constr = self.constraints[c_i]
-            las = [la[p_i] for p_i in posy_idxs]
-            nus = [[nu[i] for i in self.m_idxs[p_i]] for p_i in posy_idxs]
-            p_var_senss = constr.sens_from_dual(las, nus)
-            # ...then add to it the constant sensitivities of each constraint
-            var_senss += p_var_senss
-            # also, add each constraint's sensitivities to the results
-            # TODO: enable this once there's a plan for how to use it
-            # result["sensitivities"]["constraints"][str(constr)] = constr_sens
+        var_senss = self.sens_from_dual(la[1:].tolist(),
+                                        [[nu[i] for i in m_idx]
+                                         for m_idx in self.m_idxs[1:]])
+        # add cost's sensitivity in
+        var_senss += {var: sum([self.cost.exps[i][var]*nu[i] for i in locs])
+                      for (var, locs) in self.cost.varlocs.items()
+                      if (var in self.cost.varlocs
+                          and var not in self.posynomials[0].varlocs)}
```

Comment:
I know this isn't new code but part of it is confusing me. This only happens when the cost is not `posynomials[0]`?